cmd/wfx/cmd/root/cmd.go

```diff
@@ -191,53 +192,17 @@ Examples of tasks are installation of firmware or other types of commands issued
 		}
 
 		// check for socket-based activation (systemd)
-		{
-			listeners, _ := activation.Listeners()
-			n := len(listeners)
-			if n > 0 {
-				if n != 2 {
-					return fmt.Errorf("invalid fd count: %d", n)
-				}
-				log.Info().Int("count", n).Msg("Adopting sockets provided by systemd")
-				south, err := createSouthboundCollection([]string{kindHTTP.String()}, storage)
-				if err != nil {
-					return fault.Wrap(err)
-				}
-				south.servers[0].Listener = listeners[0] // use listener created by systemd
-				serverCollections = append(serverCollections, south)
-
-				north, err := createNorthboundCollection([]string{kindHTTP.String()}, storage)
-				if err != nil {
-					return fault.Wrap(err)
-				}
-				north.servers[0].Listener = listeners[1] // use listener created by systemd
-				serverCollections = append(serverCollections, north)
-			}
-		}
+		listeners, _ := activation.Listeners()
+		serverCollections = append(serverCollections, adoptListeners(listeners, storage)...)
 
 		errChan := make(chan error)
 		for _, collection := range serverCollections {
 			for i := range collection.servers {
 				// capture loop variable
```

Comment:
... i.e. making a closure out of this. Maybe put a note here to refactor it when Go supports auto-closure'ing of locals  and distros have catched up?

Reply:
Yep, I will add a note